common/djangoapps/student/helpers.py

```diff
@@ -354,3 +354,18 @@ def get_spam_name(email):
         # Translators: Other email Spam folder translation
         spam_name = pgettext('other_email_providers', 'Spam')
     return spam_name
+
+
+def is_origin_subdomain_url(request, origin):
+    host = request.META['HTTP_HOST']
+
+    def get_toplevel_domain(url):
+        try:
+            start_index = url.rindex('.', 0, url.rindex('.') - 1)
+            start_index = start_index + 1
+        except:
+            start_index = 0
+
+        return url[start_index:]
+
+    return get_toplevel_domain(host) == get_toplevel_domain(origin)
```

Comment:
why dont you use `url = urlparse.urlparse(origin)` and use `url.hostname` to see if the domain name is the same? Refer back to [urlparse](https://docs.python.org/2/library/urlparse.html)

Reply:
because in this situation I am only comparing to check if the top-level domain `.edraak.org` matches.
come to think of it it might not be the must secure way to go, we could instead check for the origin in `CORS_ORIGIN_WHITELIST` from the setting, what do you think ?